### PR TITLE
Add ability to reconfigure v2 project rules

### PR DESCRIPTION
**Pull request by johnnyd-eth**
## What does this PR do and why?

Tx still not working (we think Rink contracts needs updating), but everything in the rules tab is loading and submitting to the tx properly. 

Other code changes:
- created component for `ProjectConfigurationFieldsContainer` to avoid duplicate `<Col md={!hidePreview ? 10 : 24} xs={24}>` (this, as well as all the tabs should be moved out of v2create and into a new directory in a follow-up PR)
- create generic component for funding/token/rules tabs

## Screenshots or screen recordings

https://user-images.githubusercontent.com/96150256/160156420-e9ca773d-cdbd-49f6-b229-58c3c0b25b4e.mp4

## Acceptance checklist

- [x] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [x] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).

### CODE REVIEW

File: src/components/v2/V2Create/FormItemLabel.tsx [hunk 1]
```diff
@@ -1,11 +1,18 @@
...
-import { PropsWithChildren, useContext } from 'react'
+import { ChildElems } from 'models/child-elems'
+import { CSSProperties, useContext } from 'react'
 
-export default function FormItemLabel({ children }: PropsWithChildren<{}>) {
+export default function FormItemLabel({
+  children,
+  style,
+}: {
+  children: ChildElems | string
```
Comment (aeolianeth) [inline src/components/v2/V2Create/FormItemLabel.tsx:9]: **issue**: we can keep PropsWithChildren like this:

`PropsWithChildren<{ style?: CSSProperties }>`

File: src/components/v2/V2Create/tabs/ProjectConfigurationFieldsContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { Col } from 'antd'
+import { ChildElems } from 'models/child-elems'
+
+export default function ProjectConfigurationFieldsContainer({
+  hidePreview,
+  children,
+}: {
+  hidePreview?: boolean
```
Comment (aeolianeth) [inline src/components/v2/V2Create/tabs/ProjectConfigurationFieldsContainer.tsx:8]: **issue**: lets use `PropsWithChildren` here

File: src/components/v2/V2Create/tabs/ProjectConfigurationFieldsContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  children?: ChildElems
+}) {
+  return (
+    <Col md={!hidePreview ? 10 : 24} xs={24}>
```
Comment (aeolianeth) [inline src/components/v2/V2Create/tabs/ProjectConfigurationFieldsContainer.tsx:12]: **praise** yeyuh

File: src/components/v2/V2Create/tabs/RulesTab/RulesTabContent.tsx [hunk 4]
```diff
@@ -37,22 +43,21 @@ export default function RulesTabContent() {
...
     },
-    [dispatch],
+    [dispatch, onFinish],
   )
 
   const resetForm = useCallback(() => {
     form.setFieldsValue({
-      pausePay: fundingCycleMetadata?.pausePay ?? false,
-      pauseMint: fundingCycleMetadata?.pauseMint ?? false,
+      pausePay: fundingCycleMetadata?.pausePay,
```
Comment (aeolianeth) [inline src/components/v2/V2Create/tabs/RulesTab/RulesTabContent.tsx:53]: **question** why this change? does `undefined` produce the same outcome? feel like we could be explicit and keep it as `false`

File: src/components/v2/V2Create/tabs/RulesTab/RulesTabContent.tsx [hunk 5]
```diff
@@ -87,36 +91,29 @@ export default function RulesTabContent() {
...
           >
-            <div style={{ display: 'flex' }}>
-              <FormItemLabel>
-                <Trans>Pause payments</Trans>
-              </FormItemLabel>
-              <Switch
-                onChange={val => form.setFieldsValue({ pausePay: val })}
-              />
-            </div>
+            <Switch />
```
Comment (aeolianeth) [inline src/components/v2/V2Create/tabs/RulesTab/RulesTabContent.tsx:103]: haha nice, good catch.

File: src/components/v2/V2Project/V2ProjectReconfigureModal/drawers/V2ReconfigureFundingDrawer.tsx [hunk 6]
```diff
@@ -1,32 +1,28 @@
...
-  onSave: VoidFunction
   onClose: VoidFunction
+  title: JSX.Element
+  content: JSX.Element
 }) {
   return (
     <Drawer visible={visible} {...drawerStyle} onClose={onClose}>
       <h3>
-        <Trans>Reconfigure funding target/duration</Trans>
+        <Trans>{title}</Trans>
```
Comment (aeolianeth) [inline src/components/v2/V2Project/V2ProjectReconfigureModal/drawers/V2ReconfigureFundingDrawer.tsx:21]: **issue**: shouldn't wrap in `Trans` tag if there's no static text

File: src/components/v2/V2Project/index.tsx [hunk 7]
```diff
@@ -54,9 +54,10 @@ export default function V2Project() {
   )
 
   const distributionLimitCurrencySymbol =
-    currencyMetadata[distributionLimitCurrency?.toNumber() ?? 1].symbol
+    currencyMetadata[distributionLimitCurrency?.toNumber() ?? 1]?.symbol
```
Comment (aeolianeth) [inline src/components/v2/V2Project/index.tsx:57]: **issue** lets use `V2CurrencySymbol` function here, and also I don't think we should fall back to ETH - we should probably just let it fail.

File: src/locales/ru/messages.po [hunk 8]
```diff
@@ -1440,16 +1440,21 @@ msgstr "Перенастроить финансирование"
...
-#: src/components/v2/V2Project/V2ProjectReconfigureModal/drawers/V2ReconfigureFundingDrawer.tsx
-msgid "Reconfigure funding target/duration"
+#: src/components/v2/V2Project/V2ProjectReconfigureModal/index.tsx
+msgid "Reconfigure funding target/duration/payouts"
 msgstr ""
 
 #: src/components/v2/V2Project/V2ProjectReconfigureModal/drawers/V2ReconfigureProjectDetailsDrawer.tsx
 #: src/components/v2/V2Project/V2ProjectReconfigureModal/index.tsx
 msgid "Reconfigure project details"
+msgstr "<<<<<<< HEAD"
```
Comment (aeolianeth) [inline src/locales/ru/messages.po:1450]: **issue**

File: src/locales/tr/messages.po [hunk 9]
```diff
@@ -1440,16 +1440,21 @@ msgstr "Finansmanı yeniden yapılandır"
...
-#: src/components/v2/V2Project/V2ProjectReconfigureModal/drawers/V2ReconfigureFundingDrawer.tsx
-msgid "Reconfigure funding target/duration"
+#: src/components/v2/V2Project/V2ProjectReconfigureModal/index.tsx
+msgid "Reconfigure funding target/duration/payouts"
 msgstr ""
 
 #: src/components/v2/V2Project/V2ProjectReconfigureModal/drawers/V2ReconfigureProjectDetailsDrawer.tsx
 #: src/components/v2/V2Project/V2ProjectReconfigureModal/index.tsx
 msgid "Reconfigure project details"
+msgstr "<<<<<<< HEAD"
```
Comment (aeolianeth) [inline src/locales/tr/messages.po:1450]: **issue**

**Review by aeolianeth — CHANGES_REQUESTED** [review]
@johnnyd-eth a few comments for ya but on the whole LGTM!

File: src/components/v2/V2Create/FormItemLabel.tsx (hunk 1 above)
Comment (johnnyd-eth) [reply src/components/v2/V2Create/FormItemLabel.tsx:9]: Oh ok nice

File: src/components/v2/V2Create/tabs/RulesTab/RulesTabContent.tsx (hunk 4 above)
Comment (johnnyd-eth) [reply src/components/v2/V2Create/tabs/RulesTab/RulesTabContent.tsx:53]: `undefined` does produce the same outcome

File: src/components/v2/V2Project/V2ProjectReconfigureModal/drawers/V2ReconfigureFundingDrawer.tsx (hunk 6 above)
Comment (johnnyd-eth) [reply src/components/v2/V2Project/V2ProjectReconfigureModal/drawers/V2ReconfigureFundingDrawer.tsx:21]: Ahh yep good catch

File: src/components/v2/V2Project/index.tsx (hunk 7 above)
Comment (johnnyd-eth) [reply src/components/v2/V2Project/index.tsx:57]: Yeah ok

**Comment by johnnyd-eth** [comment]
Thanks @aeolianeth I've addressed all those comments. 

Also added some further changes:
- 'Reconfigure upcoming' button on `V2FundingCycleSection` -> uses V2ReconfigureModal but hides projectDetails 
- add 'Owner can token minting' to FC card
- fix bug in `RulesTabContent` where `pauseMint` was getting the opposite value to what was specified in the 'Allow token minting' field

File: src/utils/v2/currency.ts [hunk 10]
```diff
@@ -30,15 +30,15 @@ const CURRENCY_MAP: { [key in V1CurrencyOption]: V2CurrencyOption } = {
 export const V2CurrencyName = (
   currency?: V2CurrencyOption,
 ): CurrencyName | undefined =>
-  currency !== undefined ? V2_CURRENCY_METADATA[currency].name : undefined
+  currency !== undefined ? V2_CURRENCY_METADATA[currency]?.name : undefined
```
Comment (aeolianeth) [inline src/utils/v2/currency.ts:33]: @johnnyd-eth presumedly if `currency` was defined then `V2_CURRENCY_METADATA[currency]` is always defined. Any particular reason for this change?

**Review by aeolianeth — APPROVED** [review]
@johnnyd-eth apppprroooovvveeeddd
